File: lift/src/main/scala/net/destinylounge/UDPClient.scala
package net.destinylounge

import java.net.{DatagramPacket, InetAddress, DatagramSocket}
import net.liftweb.common.Logger
import code.lib.ConfigurationManager

/**
 * Created by IntelliJ IDEA.
 * User: markhalldev
 * Date: 7/7/11
 * Time: 9:13 PM
 * To change this template use File | Settings | File Templates.
 */

object UDPClient extends Logger {
//  val server = "localhost"
  // ToDo MEH Get values from config
  val server = ConfigurationManager.getSetting("dvm.ip").toString
  val port = 2638

  def sendMessage(message : String) {
     var clientSocket: DatagramSocket = new DatagramSocket
     var IPAddress: InetAddress = InetAddress.getByName(server)
     var sendData: Array[Byte] = new Array[Byte](1024)
     var receiveData: Array[Byte] = new Array[Byte](1024)

     debug("Sending message:" + message)
     sendData = message.getBytes
     var sendPacket: DatagramPacket = new DatagramPacket(sendData, sendData.length, IPAddress, port)
     clientSocket.send(sendPacket)
     var receivePacket: DatagramPacket = new DatagramPacket(receiveData, receiveData.length)
     clientSocket.receive(receivePacket)
     var modifiedSentence: String = new String(receivePacket.getData)
     debug("FROM SERVER:" + modifiedSentence)
     clientSocket.close
   }

}

File: src/java/net/destinylounge/UDPClient.java
package net.destinylounge;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.DatagramPacket;
import java.net.DatagramSocket;
import java.net.InetAddress;

/**
 * Created by IntelliJ IDEA.
 * User: markhalldev
 * Date: 7/7/11
 * Time: 7:09 PM
 * To change this template use File | Settings | File Templates.
 */
class JavaUDPClient
{
   public static void main(String args[]) throws Exception
   {
       BufferedReader inFromUser = new BufferedReader(new InputStreamReader(System.in));
      DatagramSocket clientSocket = new DatagramSocket();
      InetAddress IPAddress = InetAddress.getByName("localhost");
      byte[] sendData = new byte[1024];
      byte[] receiveData = new byte[1024];
      try {
          while(true) {
              String sentence = inFromUser.readLine();
              sendData = sentence.getBytes();
              DatagramPacket sendPacket = new DatagramPacket(sendData, sendData.length, IPAddress, 9876);
              clientSocket.send(sendPacket);
              DatagramPacket receivePacket = new DatagramPacket(receiveData, receiveData.length);
              clientSocket.receive(receivePacket);
              String modifiedSentence = new String(receivePacket.getData());
              System.out.println("FROM SERVER:" + modifiedSentence);
          }
      } finally {
        clientSocket.close();
      }
   }
}
